purchases/src/main/kotlin/com/revenuecat/purchases/utils/CoilImageDownloader.kt

```diff
@@ -0,0 +1,36 @@
+package com.revenuecat.purchases.utils
+
+import android.content.Context
+import android.net.Uri
+import coil.ImageLoader
+import coil.disk.DiskCache
+import coil.request.ImageRequest
+
+private const val MAX_CACHE_SIZE_BYTES = 25 * 1024 * 1024L // 25 MB
+private const val PAYWALL_IMAGE_CACHE_FOLDER = "revenuecatui_cache"
+
+class CoilImageDownloader(
+    private val applicationContext: Context,
+) {
+    fun downloadImage(uri: Uri) {
+        val request = ImageRequest.Builder(applicationContext)
+            .data(uri)
+            .build()
+        applicationContext.getRevenueCatUIImageLoader().enqueue(request)
+    }
+}
+
+/**
+ * This downloads paywall images in a specific cache for RevenueCat.
+ * If you update this, make sure the version in the [RemoteImage] composable is also updated.
```

Comment:
This is not ideal, but I don't want to expose this or the relevant properties if possible, so I thought it was ok to duplicate this bit.